fbw-common/src/systems/instruments/src/ND/shared/RwyAheadAdvisory.tsx

```diff
@@ -0,0 +1,58 @@
+import { DisplayComponent, EventBus, FSComponent, Subject, VNode } from '@microsoft/msfs-sdk';
+import { Arinc429Word } from '@flybywiresim/fbw-sdk';
+import { FmsOansData } from '@flybywiresim/oanc';
+import { RopRowOansSimVars } from '../../MsfsAvionicsCommon/providers';
+
+export interface RwyAheadAdvisoryProps {
+  bus: EventBus;
+}
+
+export class RwyAheadAdvisory extends DisplayComponent<RwyAheadAdvisoryProps> {
+  private readonly flagRef = FSComponent.createRef<SVGTextElement>();
+
+  private readonly qfu = Subject.create<string>('');
+
+  onAfterRender(node: VNode) {
+    super.onAfterRender(node);
+
+    const sub = this.props.bus.getSubscriber<RopRowOansSimVars & FmsOansData>();
+
+    sub
+      .on('oansWord1Raw')
+      .whenChanged()
+      .handle((it) => {
+        const w = new Arinc429Word(it);
+        this.flagRef.instance.style.display = w.getBitValueOr(11, false) ? 'block' : 'none';
+      });
```

Comment:
We should try to avoid this old imperative style for new framework code, and instead use ConsumerSubjects, MappedSubjects, and use subscribables directly in the style/class attributes.

(this applies to multiple new files in the PR)

Reply:
Yes, it's been a learning curve. In my latest code I also try to avoid this style. I improved two of those new files, can't improve all unfortunately.